File: src/main/java/io/github/jho951/assignment/job/web/dto/ImageJobCreateRequest.java
package io.github.jho951.assignment.job.web.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;

public record ImageJobCreateRequest(
        @NotBlank(message = "imageUrl is required")
        @Pattern(regexp = "^https?://.+$", message = "imageUrl must be a valid http/https URL")
        String imageUrl
) {}

File: src/main/kotlin/io/github/jho951/assignment/job/web/dto/ImageJobCreateRequest.kt
package io.github.jho951.assignment.job.web.dto

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Pattern
import kotlin.jvm.JvmRecord

@JvmRecord
data class ImageJobCreateRequest(
    @field:NotBlank(message = "imageUrl is required")
    @field:Pattern(regexp = "^https?://.+$", message = "imageUrl must be a valid http/https URL")
    val imageUrl: String?
)
